src/solvers/flattening/pointer_logic.cpp

```diff
@@ -104,6 +104,23 @@ exprt pointer_logict::pointer_expr(
   // https://gcc.gnu.org/onlinedocs/gcc-4.8.0/gcc/Pointer-Arith.html
   if(subtype.id() == ID_empty)
     subtype = char_type();
+  if(
+    object_expr.id() == ID_string_constant &&
+    object_expr.type().id() == ID_array)
```

Comment:
I suspect if it's a string_constant then the array is an invariant?

Reply:
Indeed, leaving that check now to an unconditional `to_array_type`.